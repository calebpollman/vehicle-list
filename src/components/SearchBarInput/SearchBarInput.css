.search-bar-input-container {
  display: flex;
  height: 100%;
  margin-left: auto;
  width: 100%;
}

.search-bar-input {
  background-color: var(--color-search-bar-background);
  border: transparent;
  color: var(--color-search-bar-text);
  font-family: var(--font-application);
  font-size: var(--font-size-medium);
  font-weight: var(--font-weight-regular);
  height: 100%;
  outline: none;
  width: 100%;
}