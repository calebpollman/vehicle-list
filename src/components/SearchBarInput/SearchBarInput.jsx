import React from 'react';
import PropTypes from 'prop-types';
import './SearchBarInput.css';

const SearchBarInput = ({updateSearchTerm}) => {
  return (
    <div className="search-bar-input-container">
      <input
        autoFocus
        className="search-bar-input"
        onChange={event => updateSearchTerm(event)}
        placeholder="Search Make, Model, Year"
      />
    </div>
  );
}

SearchBarInput.propTypes = {
  updateSearchTerm: PropTypes.func,
}

export default SearchBarInput;