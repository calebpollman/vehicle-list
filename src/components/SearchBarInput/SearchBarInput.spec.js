import React from 'react';
import {shallow} from 'enzyme';
import SearchBarInput from './SearchBarInput';

describe('SearchBarInput', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<SearchBarInput />);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  
  it('renders a wrapper <div /> and an <input />', () => {
    expect(wrapper.find('div').length).toEqual(1);
    expect(wrapper.find('input').length).toEqual(1);
  });
});