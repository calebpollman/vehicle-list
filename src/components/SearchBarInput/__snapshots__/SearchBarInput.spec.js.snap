// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBarInput should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchBarInput />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <input
        autoFocus={true}
        className="search-bar-input"
        onChange={[Function]}
        placeholder="Search Make, Model, Year"
/>,
      "className": "search-bar-input-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "autoFocus": true,
        "className": "search-bar-input",
        "onChange": [Function],
        "placeholder": "Search Make, Model, Year",
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <input
          autoFocus={true}
          className="search-bar-input"
          onChange={[Function]}
          placeholder="Search Make, Model, Year"
/>,
        "className": "search-bar-input-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "autoFocus": true,
          "className": "search-bar-input",
          "onChange": [Function],
          "placeholder": "Search Make, Model, Year",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
